pkg/datapath/linux/devices.go

```diff
@@ -141,23 +150,15 @@ func (dm *DeviceManager) Detect() error {
 				option.Config.IPv6MCastDevice = k8sNodeDev
 				log.WithField(option.IPv6MCastDevice, option.Config.IPv6MCastDevice).Info("IPv6 multicast device detected")
 			} else {
-				return fmt.Errorf("Unable to determine Multicast device. Use --%s to specify it",
+				return nil, fmt.Errorf("unable to determine Multicast device. Use --%s to specify it",
 					option.IPv6MCastDevice)
 			}
 		}
 	}
 
-	option.Config.Devices = dm.getDevices()
-	log.WithField(logfields.Devices, option.Config.Devices).Info("Detected devices")
-
-	return nil
-}
-
-// GetDevices returns the current list of devices Cilium should attach programs to.
-func (dm *DeviceManager) GetDevices() []string {
-	dm.Lock()
-	defer dm.Unlock()
-	return dm.getDevices()
+	deviceList := dm.getDevices()
+	log.WithField(logfields.Devices, deviceList).Info("Detected devices")
```

Comment:
Hmm, now I look around at the logs it seems like we're sprouting multiple of this one. Maybe remove this one?

Reply:
I'd keep both this one in `Detect()` and the one in `Listen()`, but make sure there's no other duplication.